### DISCUSSION TITLE
Autocompletion and auto-suggestions in cli?

### DISCUSSION THREAD

**Discussion by Animeshz** [PaddiM8/kalk#76] [Q&A] [answered]
Is it a bug or intentional, that web-based kalk do changes `sqrt` `integral` `sum` etc and also closes the parenthesis itself while cli doesn't?

**Comment by PaddiM8** [comment, marked as answer]
The cli version does this as well, but you have to press tab. At the time at least, I couldn't find a way to do it automatically with the rust crate used to deal with cli input/output (`rustyline`), so implementing this may be difficult. It would be really nice though.

**Reply by Animeshz** [reply]
Oh alright! I'm not sure about the internals, but I guess you could use https://crates.io/crates/console for listening to live key-presses, simplest example is [here](https://github.com/mitsuhiko/dialoguer/blob/1b84eaf82406f0d64c902245cdf270b16bd823ad/src/prompts/confirm.rs#L177-L203), this can either detect if enter is pressed or any char by matching with `Key::Enter` and `Key::Char(char)` respectively.

Edit: if some special keyword appear you can [clear_chars](https://docs.rs/console/0.15.0/console/struct.Term.html#method.clear_chars) for the length and write those special characters or autocomplete the braces.

**Reply by PaddiM8** [reply]
Hmm might be doable somehow if it integrates well with rustyline. Will have to poke around and see.
